src/objects/aff/zcl_abapgit_json_path.clas.locals_imp.abap

```diff
@@ -35,11 +40,165 @@ CLASS lcl_json_path DEFINITION CREATE PUBLIC.
+
+    IF path_contains_array( lv_path ) = abap_true.
+
+      lt_path_elements = get_path_elements( lv_path ).
+
+      build_json( EXPORTING it_path_elements = lt_path_elements
+                            iv_value         = lv_value
+                  CHANGING  cv_json_string   = lv_json ).
+
+      ro_result = zcl_abapgit_ajson=>parse( lv_json ).
+    ELSE.
+
+      REPLACE FIRST OCCURRENCE OF '$.' IN lv_path WITH ''.
+      REPLACE '.' IN lv_path WITH '/'.
+      ro_result = zcl_abapgit_ajson=>create_empty( iv_keep_item_order = abap_true ).
+      ro_result->set( iv_path = lv_path
+                      iv_val  = lv_value ).
+    ENDIF.
+
+
+  ENDMETHOD.
+
+  METHOD path_contains_array.
+    DATA lv_array_pattern TYPE string VALUE `.*\[.*\].*`.
+    rv_result = boolc( matches( val   = iv_path
+                                regex = lv_array_pattern ) ).
+  ENDMETHOD.
+
+  METHOD build_json.
+    DATA: lt_new_path_element TYPE string_table,
+          lv_sub_match        TYPE string,
+          lv_key_name         TYPE string,
+          lv_key_value        TYPE string,
+          lv_name             TYPE string,
+          lv_first_elem       TYPE string.
+
+    lt_new_path_element = it_path_elements.
+
+    IF lines( lt_new_path_element ) = 0.
+      RETURN.
+    ENDIF.
+
+    READ TABLE lt_new_path_element INTO lv_first_elem INDEX 1.
+
+    IF lv_first_elem = `$`. " is root level
+
+      DELETE lt_new_path_element INDEX 1.
+      build_json( EXPORTING it_path_elements = lt_new_path_element
+                            iv_value         = iv_value
+                  CHANGING  cv_json_string   = cv_json_string ).
+
+    ELSEIF is_primitiv( lv_first_elem ) = abap_true.
+
+      cv_json_string = cv_json_string && | \{"{ lv_first_elem+1 }": |.
+
+      DELETE lt_new_path_element INDEX 1.
+
+      build_json( EXPORTING it_path_elements = lt_new_path_element
+                            iv_value         = iv_value
+                  CHANGING  cv_json_string   = cv_json_string ).
+
+      cv_json_string = cv_json_string && ` }`.
+
+    ELSE. " is array
+
+      FIND REGEX `\[(.*)\]` IN lv_first_elem SUBMATCHES lv_sub_match.
+      FIND REGEX `(\w+)(?==='([^']*)')` IN lv_sub_match SUBMATCHES lv_key_name lv_key_value.
+      READ TABLE lt_new_path_element INTO lv_name INDEX 2.
+
+
+      DELETE lt_new_path_element INDEX 1.
+      DELETE lt_new_path_element INDEX 1.
+
+      IF lines( lt_new_path_element ) = 0.
+
+        cv_json_string = cv_json_string &&
+          |[ \{ "{ lv_key_name }": "{ lv_key_value }", "{ lv_name+1 }": "{ iv_value }"\} ]|.
+
+      ELSE.
+
+        cv_json_string = cv_json_string && |[ \{ "{ lv_key_name }": "{ lv_key_value }", "{ lv_name+1 }":|.
+
+        build_json( EXPORTING it_path_elements = lt_new_path_element
+                              iv_value         = iv_value
+                    CHANGING  cv_json_string   = cv_json_string ).
+
+        cv_json_string = cv_json_string && `} ] `.
```

Comment:
Minor: I guess I would have moved the whole array handling  to a dedicated method (maybe even with sub methods).